### PR TITLE
influxdb batching filter

### PR DESCRIPTION
**Pull request by dfuentes**
Simple influxdb batching filter.  Currently batches only on time, not number of messages.

### CODE REVIEW

**Comment by dfuentes** [comment]
@mohit the influx batcher we talked about

File: lua/filters/influxdbbatch.lua [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+
+    [{"points":[[1.409378221e+21,"log","test","systemName","TcpInput",5,1,"test"]],"name":"heka.MyLogger","columns":["Time","Type","Payload","Hostname","Logger","Severity","syslogfacility","programname"]}]
+
+--]=]
+
+require "cjson"
+require "string"
+require "table"
+
+local series_orig  = read_config("series") or "series"
```
Comment (mohit) [inline lua/filters/influxdbbatch.lua]: don't think default should be series. Maybe make it a required arg?

File: lua/filters/influxdbbatch.lua [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+    Type = true,
+    Payload = true,
+    Hostname = true,
+    Pid = true,
+    Logger = true,
+    Severity = true,
+    EnvVersion = true
+}
+
+local base_fields_list = {
```
Comment (mohit) [inline lua/filters/influxdbbatch.lua]: I don't think we need both of them, you can just go over the map only looking at the keys

File: lua/filters/influxdbbatch.lua [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+            return val
+        end
+        return "%{"..key.."}"
+    end
+end
+
+-- Remove blacklisted fields from the set of base fields that we use, and
+-- create a table of dynamic fields to skip.
+local used_base_fields = {}
+local skip_fields_str = read_config("skip_fields")
```
Comment (mohit) [inline lua/filters/influxdbbatch.lua]: keep all the config reading in the same place?

File: lua/filters/influxdbbatch.lua [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+local skip_fields = {}
+if skip_fields_str then
```
Comment (mohit) [inline lua/filters/influxdbbatch.lua]: make this a function?

**Comment by dfuentes** [comment]
Talked with @mohit offline, going to go ahead with merge.  Comments above are on code which was copied from influx schema encoder from heka-private.

**Pull request merged**